### ISSUE TITLE
ContactData.equals() might cause NullPointerException

### ISSUE THREAD

**Issue by alexsaalfeld** [blainepwnz/AndroidContacts#19]
The equals Method of ContactData is currently implemented in a way it causes a NullPointerException when one of the fields is null.
Please use `Objects.equals(a,b)` instead of `a.equals(b)`.

```
   @Override
   public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ContactData contact = (ContactData) o;

        if (contactId != contact.contactId) return false;
        if (!emailList.equals(contact.emailList)) return false;
        if (!phoneList.equals(contact.phoneList)) return false;
        if (!addressesList.equals(contact.addressesList)) return false;
        if (!websitesList.equals(contact.websitesList)) return false;
        if (!imAddressesList.equals(contact.imAddressesList)) return false;
        if (!relationsList.equals(contact.relationsList)) return false;
        if (!specialDatesList.equals(contact.specialDatesList)) return false;
        if (!groupList.equals(contact.groupList)) return false;
        if (!note.equals(contact.note)) return false;
        if (!nickName.equals(contact.nickName)) return false;
        if (!sipAddress.equals(contact.sipAddress)) return false;
        if (!photoUri.equals(contact.photoUri)) return false;
        if (!organization.equals(contact.organization)) return false;
        if (!nameData.equals(contact.nameData)) return false;
        if (isFavorite != contact.isFavorite) return false;
        return compositeName.equals(contact.compositeName);

    }
```
Thank you!

**Comment by blainepwnz** [comment]
All of those fields are null protected but for compositeName, it can throw NPE if only compositeName was set null explicitly, anyway going to fix it

**Comment by blainepwnz** [comment]
fixed in #20
